samza-api/src/main/java/org/apache/samza/table/AsyncReadWriteTable.java

```diff
@@ -90,6 +91,27 @@
    */
   CompletableFuture<Void> putAllAsync(List<Entry<K, V>> entries, Object ... args);
 
+  /**
+   * Asynchronously updates an existing record for a given key with the specified update.
+   *
+   * @param key the key with which the specified {@code value} is to be associated.
+   * @param update the update applied to the record associated with a given {@code key}.
+   * @param args additional arguments
+   * @throws NullPointerException if the specified {@code key} is {@code null}.
+   * @return CompletableFuture for the operation
+   */
+  CompletableFuture<Void> updateAsync(K key, U update, Object ... args);
+
+  /**
+   * Asynchronously updates the existing records for the given keys with their corresponding updates.
+   *
+   * @param updates the key and update mappings.
+   * @param args additional arguments
+   * @throws NullPointerException if any of the specified {@code entries} has {@code null} as key.
+   * @return CompletableFuture for the operation
+   */
+  CompletableFuture<Void> updateAllAsync(List<Entry<K, U>> updates, Object ... args);
```

Comment:
Can you change the return type to CompletionStage instead of future?
we don't want to accidentally promote folks to fallback to synchronous programming by using .get() or join on this. Please apply the same comment for all the interfaces.

Reply:
Hey, I used `CompletableFuture` just to be in line with the existing code (putAsync, getAsync, etc.).

Reply:
@mynameborat : this patch is getting quite huge so far. Let's have another patch if we want to change all the return type for table methods. Let's keep the scope to be partial update api for this patch. Thanks.